Feature/ASAA-20 Commit Screen Branch Info Additions

LuisRocketRangel opened the pull request:
ASAA-20 Add Branch Info to Commit Screen 

Requirements:
-ADd Branch Info to Commit Screen UI

Solution:
-Modify API call to get branch info as well as commit info
-Add UI element to display branch name
-Add branch name list to the commit screen state

LuisRocketRangel committed:
ASAA-20 Added Branches to Commit Items

Requirements:
-Add Branch names to the commit UI

Solution:
-Added API call to get branches in a Repo
-Added API call to get commits for a branch
-Updated the Commit Item Compose to include Branch info
-Made BranchName and Hash bold, as per Zeplin
-Added BranchDto and updated CommitDto to have branch data

LuisRocketRangel committed:
ASAA-20 Added Branch Names to the Commit Screen State

Requirements:
-Add access to Branch names for the commit Screen UI

Solution:
-BranchName list to the CommitViewModel

LuisRocketRangel committed:
Merge remote-tracking branch 'origin/develop' into feature/ASAA-20

# Conflicts:
#	compose/src/main/java/com/bottlerocketstudios/compose/repository/RepositoryCommitScreen.kt

LuisRocketRangel committed:
ASAA-20 Merge with Develop

Requirements:
-Merge

Solution:
-Fix Detekt issues
-Clean commit screen

BottleRocket-Colin commented on app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/repository/RepositoryCommitViewModel.kt:
```diff
@@ -47,27 +47,43 @@ class RepositoryCommitViewModel : BaseViewModel() {
-                val result = repo.getCommits(slug, repoName)
-                when (result) {
-                    is Status.Success -> srcCommits.value = result.data
+                when (val branchCallResult = repo.getBranches(slug, repoName)) {
```
Look into using handlingErrors extension function here.

LuisRocketRangel replied:
Donzo!

BottleRocket-Colin commented on compose/src/main/java/com/bottlerocketstudios/compose/repository/RepositoryCommitScreen.kt:
```diff
@@ -79,26 +79,29 @@ fun CommitItem(item: RepositoryCommitItemUiModel) {
-                    text = item.hash,
-                    style = MaterialTheme.typography.h5
+                    text = item.branchName,
+                    style = MaterialTheme.typography.h5.copy(fontWeight = FontWeight.Bold),
```
We have a toBold() extension function that performs this copy.

BottleRocket-Colin commented on compose/src/main/java/com/bottlerocketstudios/compose/repository/RepositoryCommitScreen.kt:
```diff
@@ -79,26 +79,29 @@ fun CommitItem(item: RepositoryCommitItemUiModel) {
                 )
+                Text(
+                    text = item.hash,
+                    style = MaterialTheme.typography.h5.copy(fontWeight = FontWeight.Bold)
```
ditto w/ respect to toBold()

BottleRocket-Colin reviewed (changes requested):
A couple of minor changes,

LuisRocketRangel committed:
ASAA-20 PR Feedback

Requirements:
-Use HandlingErrors extension for repo calls
-Use bold() extension for making compose text bold

Solution:
-Used the HandlingFunction to make the getRepoCommits call more concise
-Used bold(), instead of copy

jaredstockton commented on app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/repository/RepositoryCommitStateConverter.kt:
```diff
@@ -8,5 +8,6 @@ import com.bottlerocketstudios.compose.repository.RepositoryCommitScreenState
     itemCount = itemCount.collectAsState(),
-    commitItems = uiModels.collectAsState()
+    commitItems = uiModels.collectAsState(),
+    branchNames = branchItems.collectAsState()
```
@LuisRocketRangel nit: Let's add a trailing comma here to allow for simpler diffs if/when new arguments are added to `RepositoryCommitScreenState` in the future

jaredstockton commented on app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/repository/RepositoryCommitViewModel.kt:
```diff
@@ -47,27 +46,36 @@ class RepositoryCommitViewModel : BaseViewModel() {
                     hash = commit.hash?.take(HASH_LENGTH) ?: "",
-                    message = commit.message ?: ""
+                    message = commit.message ?: "",
+                    branchName = commit.branchName ?: ""
```
@LuisRocketRangel nit: Trailing comma :)

jaredstockton commented on data/src/main/java/com/bottlerocketstudios/brarchitecture/data/model/CommitDto.kt:
```diff
@@ -15,5 +15,6 @@ data class CommitDto(
     @Json(name = "author") val author: AuthorDto?,
-    @Json(name = "repository") val commitRepository: CommitRepositoryDto?
+    @Json(name = "repository") val commitRepository: CommitRepositoryDto?,
+    var branchName: String?
```
@LuisRocketRangel We should avoid adding additional data to Dto classes, even moreso mutable data. Going to work on an alternate solution and push up a review commit later today

jaredstockton reviewed (changes requested):
@LuisRocketRangel Working on a review commit with some additional changes here

jaredstockton committed:
ASAA-20 Review commit: Remove additional mutable branch property from CommitDto and create data class wrapper to hold onto CommitDto+Branch for internal ViewModel use

LuisRocketRangel committed:
Merge branch 'develop' into feature/ASAA-20
